examples/java/spring-boot/src/test/java/uk/gov/api/springboot/repositories/EmptyMetadataRepositoryTest.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.api.springboot.repositories;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.Test;
+
+class EmptyMetadataRepositoryTest {
+
+  @Test
+  void findAllReturnsEmptyList() {
+    EmptyMetadataRepository repository = new EmptyMetadataRepository();
+
+    assertThat(repository.findAll()).isEmpty();
+  }
+
+  @Test
+  void findAllReturnsUnmodifiableList() {
+    EmptyMetadataRepository repository = new EmptyMetadataRepository();
+
+    var metadataDaos = repository.findAll();
+
+    assertThatThrownBy(() -> metadataDaos.add(null))
```

Comment:
I generally prefer that we use `actual`:

```suggestion
    var actual = repository.findAll();

    assertThatThrownBy(() -> actual.add(null))
```